### PR TITLE
Fix curve25519 endianness

### LINKED ISSUE #84: cv25519 encryption key doesn't show as moved to card
Sent all subkeys to the card with `keytocard` but gpg --card-status gives me this:
```
sub  ed25519/4581665975B9EFDC 2022-09-29 Antonio Murdaca <antoniomurdaca@gmail.com>
sec#  ed25519/4A51A50353E8131B  created: 2022-09-29  expires: 2024-09-28
ssb>  ed25519/4581665975B9EFDC  created: 2022-09-29  expires: 2023-09-29
                                card-no: 0000 A0103125
ssb   cv25519/2213607CC8C5F771  created: 2022-09-29  expires: 2023-09-29
ssb>  ed25519/4C25A83CB3E806C1  created: 2022-09-29  expires: 2023-09-29
                                card-no: 0000 A0103125
```
basically the encryption key seems to be still on the system (verified with gpg -K too) but if I try to resend the enc key to the card gpg tells me it's alrady there

### PR DESCRIPTION
**Pull request by sosthene-nitrokey**
GPG doesn't exactly implement X25519, because it stores ECC scalars in Big-Endian format when the standardised X25519 operates with little endian scalars Therfore we need to reverse the endianness of the private key on key import.

This also highlight that the gpg-import test for X25519 was incorrect because the key import failed silently

Closes #84

### CODE REVIEW

**Comment by sosthene-nitrokey** [comment]
Thanks @hko-s and @wiktor-k for help finding the bug and the fix!

**Comment by wiktor-k** [comment]
No problem, happy to help! :+1: 

Very cool to see you working on that and also using Trussed... What a time to be alive! :partying_face: 

Have a nice day! :wave:

**Comment by runcom** [comment]
is there any way for me to build and test this on my nk3m? :) (instructions mainly)

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,12 +17,14 @@ SPDX-License-Identifier: CC0-1.0
 
 - Fix the length of the Digital signature counter DO 0x93 ([#76][])
 - PSO:CDS: Increment the signature counter ([#78][])
+- Fix endianness of curve25519 key impor([#89][])
```
Comment (robin-nitrokey) [inline CHANGELOG.md:20]: import

**Review by robin-nitrokey — APPROVED** [review]
What a nasty issue, thanks for the fix!

File: CHANGELOG.md (hunk 1 above)
Comment (sosthene-nitrokey) [reply CHANGELOG.md]: Fixed by d8c32fb

**Comment by robin-nitrokey** [comment]
@runcom We plan to release a new alpha version probably next week.  If you want to give it a try with a virtualized smartcard instead of the hardware, you can install the build dependencies (currently being added in https://github.com/Nitrokey/opcard-rs/pull/88) and run the [`virtual`](https://github.com/Nitrokey/opcard-rs/blob/main/examples/virtual.rs) example with `cargo run --features virtual --example virtual`.

**Pull request merged**